### PR TITLE
Feat/added menu item details activity

### PR DESCRIPTION
**Pull request by DFraczek**
# Pull Request: Menu Item Detail Activity Implementation

## Overview

This pull request implements the **Menu Item Detail Activity** in our Android application. This activity displays detailed information about a selected menu item, including an image, name, description, and an "Add to Basket" button.

## Changes Made

- **Created `MenuItemDetailActivity`**:
  - This activity is responsible for displaying the details of a selected menu item.
  - Utilizes a toolbar for navigation and back functionality.
  
- **User Interface**:
  - Added an `ImageView` to display the dish's image using Glide for image loading.
  - Added `TextView`s for displaying the dish name and description.
  - Added a `Button` labeled "Add to basket" for adding the item to the user's basket.

- **Navigation**:
  - Implemented navigation from the menu to the detail activity.
  - Added back navigation functionality using the toolbar.

- **Data Handling**:
  - Retrieved the `MenuItem` object passed via intent to populate the UI elements.
  - Implemented a click listener for the "Add to Basket" button, logging the action (to be expanded with actual basket logic in future updates).

## Screenshots

image

## Future Work

- Implement the functionality to add the selected menu item to the user's basket.

### CODE REVIEW

**Comment by DFraczek** [comment]
image
image

File: Android/app/src/main/java/com/example/android/activities/MenuActivity.java [hunk 1]
```diff
@@ -39,10 +39,21 @@ protected void onCreate(Bundle savedInstanceState) {
     private void initializeRecyclerView() {
         recyclerView = findViewById(R.id.recyclerViewMenuItems);
         recyclerView.setLayoutManager(new LinearLayoutManager(this));
-        menuItemAdapter = new MenuItemAdapter(null, "");
+        menuItemAdapter = new MenuItemAdapter(null, "", new MenuItemAdapter.OnItemClickListener() {
```
Comment (blackspyek) [inline Android/app/src/main/java/com/example/android/activities/MenuActivity.java:52]: extract method

**Review by blackspyek — CHANGES_REQUESTED** [review]

File: Android/app/src/main/java/com/example/android/activities/MenuItemDetailActivity.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    @Override
+    public boolean onSupportNavigateUp() {
+        finish();
+        return true;
+    }
+
+    private void displayMenuItemDetails(MenuItem menuItem) {
+        ImageView itemImage = findViewById(R.id.itemImage);
```
Comment (blackspyek) [inline Android/app/src/main/java/com/example/android/activities/MenuItemDetailActivity.java]: This imo should be extracted cause its field init

File: Android/app/src/main/java/com/example/android/activities/MenuItemDetailActivity.java [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+    private void displayMenuItemImage(MenuItem menuItem, ImageView itemImage) {
+        Glide.with(this)
+                .load(menuItem.getPhotoUrl())
+                .into(itemImage);
+    }
+
+    private void initializeAddToBasketButton() {
+        Button addToBasketButton = findViewById(R.id.addToBasketButton);
+        addToBasketButton.setOnClickListener(new View.OnClickListener() {
```
Comment (blackspyek) [inline Android/app/src/main/java/com/example/android/activities/MenuItemDetailActivity.java]: Extract

File: Android/app/src/main/java/com/example/android/models/MenuItem.java [hunk 4]
```diff
@@ -1,10 +1,17 @@
...
+import java.io.Serializable;
+
+public class MenuItem implements Serializable {
     private String photoUrl;
     private String name;
     private Double price;
     private boolean available;
+    private String description;
+
+    public String getDescription() {
```
Comment (blackspyek) [inline Android/app/src/main/java/com/example/android/models/MenuItem.java]: When you are doing getters and setters use annotations @Gettet and @Setter also for args and noargs constructor

**Review by blackspyek — CHANGES_REQUESTED** [review]

**Review by blackspyek — APPROVED** [review]
LGTM

**Pull request merged**